lib/board.rb

```diff
@@ -0,0 +1,73 @@
+require './lib/ship'
+require './lib/cell'
+require './lib/coordinates'
+
+class Board
+  attr_reader :cells
+
+  def initialize(size = 4)
+    @cells = {}
+    @size = size
+
+    (1..@size).each do |row|
+      x_coord = (row + 64).chr
+      (1..@size).each do |column|
+        coordinate = "#{x_coord}#{column}"
+        @cells[coordinate] = Cell.new(coordinate)
+      end
+    end
+  end
+
+  def valid_placement?(ship, coords)
+    ship.length == coords.length &&
+      Coordinates.consecutive?(coords) &&
+      Coordinates.not_diagonal?(coords) &&
+      not(overlap?(coords))
+  end
+
+  def place(ship, coords)
+    cells = @cells.values
+    cells.each do |cell|
+      coords.each do |coord|
+        if cell.coordinate == coord
+          cell.place_ship(ship)
+        end
+      end
+    end
+  end
+
+  def overlap?(coords)
+    values = @cells.values
+    cells_with_ships = values.find_all do |cell|
+      cell if !cell.empty?
+    end
+
+    cells_with_ships.any? do |cell|
+      coords.include?(cell.coordinate)
+    end
+  end
```

Comment:
Let's consider moving this into the Coordinates class. We can pass in the board's cells like so
`Coordinates.intersects?(@cells, coordinates)`
That means we will have to refactor this class, and Coordinates class as well. What do you think?

Reply:
I think that works. It'll be good to leverage the intersects? method in the Coordinates class to make the Board class concise. Great catch.